feat: product tours UI

ordehi opened the pull request:
## Product tours management UI

Adds a management UI for product tours, following the surveys pattern. The backend and toolbar editor already exist, this PR adds the in-app management experience.

<img width="1550" height="830" alt="CleanShot 2025-12-09 at 23 37 16@2x" src="https://github.com/user-attachments/assets/7c36d3df-8755-4638-bc56-071c09a28e19" />

<img width="2920" height="1788" alt="CleanShot 2025-12-09 at 23 38 15@2x" src="https://github.com/user-attachments/assets/f17a6b26-64b3-4650-ad01-821eafdb04de" />

<img width="1690" height="1792" alt="CleanShot 2025-12-09 at 23 38 39@2x" src="https://github.com/user-attachments/assets/155cb790-f6ba-4649-b451-4968405294c6" />

ordehi committed:
feat(product-tours): Add product tours management UI

- Add scene registration for ProductTours and ProductTour scenes
- Create list page with table, search, and status filtering
- Create single tour view with stats summary and targeting settings
- Add sidebar navigation gated by product-tours-2025 feature flag
- Add ProductTour to AccessControlResourceType enum

ordehi committed:
feat(product-tours): Add edit mode with feature flag targeting

- Add targeting_flag_filters to backend serializer for user targeting
- Create ProductTourEdit component with FeatureFlagReleaseConditions
- Refactor ProductTour to toggle between view/edit modes like surveys
- Add URL sync for edit mode (?edit=true)
- Simplify ProductTourView to read-only overview with Edit button

ordehi committed:
feat(product-tours): Add date filter, stats display, and display conditions

- Add unified date filter that applies to both stats and funnel
- Date filter defaults to tour's start_date
- Stats now show unique users as primary metric with total in parentheses
- Add buildHogQLDateFilter utility for consistent date handling
- Update display conditions to use FeatureFlagReleaseConditions (readOnly)
  matching the surveys pattern
- Fix management command to create internal_targeting_flag for tours

Tests: pre-existing TypeScript error in error_tracking (unrelated)
Build: pending
Lint: passed

ordehi committed:
fix(product-tours): Add type annotations to fix mypy errors

ordehi committed:
feat(product-tours): Edit in toolbar opens URL picker modal

Instead of navigating to toolbar launch page, show modal with
AuthorizedUrlList to pick URL for toolbar launch.

adboio commented on frontend/src/scenes/product-tours/ProductTour.tsx:
```diff
@@ -0,0 +1,45 @@
+import { BindLogic, useActions, useValues } from 'kea'
```
would it be a huge pain in the ass to move these to the top-level `posthog/products` directory? 

see example - https://github.com/PostHog/posthog/tree/master/products/error_tracking/frontend

ordehi committed:
fix(product-tours): Add missing AuthorizedUrlList import

adboio commented on posthog/management/commands/generate_random_product_tours.py:
```diff
@@ -0,0 +1,524 @@
+            # Simulate user progressing through steps
+            # Random drop-off: some complete all, some drop off at various steps
+            completion_probability = random.random()
+            steps_to_complete = num_steps if completion_probability > 0.3 else random.randint(0, num_steps - 1)
```
shut up greptile

adboio commented on frontend/src/scenes/product-tours/components/ProductToursTable.tsx:
```diff
@@ -0,0 +1,303 @@
+                              {
+                                  title: 'Status',
+                                  width: 100,
+                                  render: function Render(_: any, tour: ProductTour) {
```
everyone does this, it's fine

ordehi committed:
feat(product-tours): Address PR review feedback

  - Update manifest.tsx for proper feature gating with
  fileSystemTypes,
    treeItemsNew, and treeItemsProducts with PRODUCT_TOURS flag
  - Remove redundant completion rate card from stats summary
  - Fix missing 'flex' class in display conditions section
  - Address SQL injection concern by escaping tour ID in HogQL
  queries
  - Use LemonSkeleton for better loading state in edit view
  - Add product_tour icon type using IconSpotlight instead of
  survey icon

adboio commented on products/product_tours/backend/api/product_tour.py:
```diff
@@ -105,8 +137,25 @@ def create(self, validated_data):
 
     @transaction.atomic
     def update(self, instance, validated_data):
```
yee haw 🤠🤠🚀🚢

ordehi commented on products/product_tours/manifest.tsx:
```diff
@@ -1,8 +1,43 @@
-import { ProductManifest } from '~/types'
+import { FEATURE_FLAGS } from 'lib/constants'
```
Yup, it did cause CI to shout at me, thanks!

adboio commented on products/product_tours/backend/api/product_tour.py:
```diff
@@ -57,12 +59,41 @@ def get_feature_flag_key(self, tour: ProductTour) -> str | None:
             return tour.internal_targeting_flag.key
         return None
 
+    def get_targeting_flag_filters(self, tour: ProductTour) -> dict | None:
```
oh nice ty 

i’m still not fully attuned to the django magic, need a couple more long rests

ordehi committed:
Merge branch 'master' into feat/pt-ui
